### PR TITLE
[Breaking change] support cron cloud build jobs

### PR DESCRIPTION
**Pull request by xingao267**
This is also a breaking change which renames `branch_regex` to
`branch_name` in the `cicd` component and restrict the value to be the
exact name of the branch. The motivation behind this are:

- Enforce a 1:1 mapping between a branch to a GCP environment, which
means changes made to only one branch should trigger jobs associated
with one GCP environment (i.e. dev, staging or prod).

- Cloud Build job triggered by Cloud Scheduler requires exactly one
branch name to be specified.

Migration instruction:
* Rename `branch_regex` to `branch_name` in `cicd` template, and restrict target
branch to be exactly one branch.

Fixes https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/105

### CODE REVIEW

File: docs/tfengine/schemas/cicd.md [hunk 1]
```diff
@@ -121,3 +147,11 @@ Type: object
 Whether or not automatic triggering from a PR/push to branch. Default to true.
 
 Type: boolean
+
+### triggers.validate.run_on_schedule
+
+Whether or not automatic triggering according a specified schedule.
```
Comment (MartinPetkov) [inline docs/tfengine/schemas/cicd.md:153]: This sentence seems a bit grammatically incorrect. Did you mean to say whether or not to automatically validate?

**Review by MartinPetkov — APPROVED** [review]

File: examples/tfengine/devops.hcl [hunk 2]
```diff
@@ -50,14 +50,21 @@ template "cicd" {
...
+    # Required to create Cloud Scheduler jobs.
+    scheduler_region = "us-central1"
+
     # Prepare and enable default triggers.
     triggers = {
       validate = {}
-      plan     = {}
-      apply    = { run_on_push = false } # Do not auto run on push to branch
+      plan     = {
+        run_on_schedule = "0 12 * * *" # Run at 12 PM EST everyday
```
Comment (umairidris) [inline examples/tfengine/devops.hcl:63]: Just so I understand, this will run on push AND run on schedule ?

**Review by umairidris — APPROVED** [review]

File: docs/tfengine/schemas/cicd.md (hunk 1 above)
Comment (xingao267) [reply docs/tfengine/schemas/cicd.md]: reworded, ptal

File: examples/tfengine/devops.hcl (hunk 2 above)
Comment (xingao267) [reply examples/tfengine/devops.hcl:63]: yea, run_on_push is default to true, so it doesn't need to be specified explicitly.

**Review by MartinPetkov — APPROVED** [review]

**Review by umairidris — APPROVED** [review]

**Pull request merged**